pkg/reconciler/tektonwrapper/tektonwrapper.go

```diff
@@ -0,0 +1,271 @@
+		// under powered
+		return reconcile.Result{}, nil
+	case v1alpha1.TektonWrapperStateInProgress:
+		//TODO even though we have the PipelineRun watch, we could list on our label and make sure the PipelineRun is
+		// active and we have not missed any events; for now, deferring on this
+		return reconcile.Result{}, nil
+	}
+
+	// get embedded PR
+	pr = &v1beta1.PipelineRun{}
+	if len(tw.Spec.PipelineRun) > 0 {
+		decodingScheme := runtime.NewScheme()
+		utilruntime.Must(v1beta1.AddToScheme(decodingScheme))
+		decoderCodecFactory := serializer.NewCodecFactory(decodingScheme)
+		decoder := decoderCodecFactory.UniversalDecoder(v1beta1.SchemeGroupVersion)
+		prHydrateErr := runtime.DecodeInto(decoder, tw.Spec.PipelineRun, pr)
+		if prHydrateErr != nil {
+			return reconcile.Result{}, prHydrateErr
+		}
+	} else {
+		msg := fmt.Sprintf("TektonWrapper %s:%s has no embedded PipelineRun", tw.Namespace, tw.Name)
+		log.Info(msg)
+		r.eventRecorder.Event(&tw, corev1.EventTypeWarning, "MissingPipelineRun", msg)
+		return reconcile.Result{}, nil
+	}
+	if pr == nil {
+		msg := fmt.Sprintf("TektonWrapper %s:%s could not produced a pipeline run with buf %#v", tw.Namespace, tw.Name, tw.Spec.PipelineRun)
+		log.Info(msg)
+		r.eventRecorder.Event(&tw, corev1.EventTypeWarning, "MissingPipelineRun", msg)
+		return reconcile.Result{}, nil
+	}
+
+	// we hydrate PipelineRun first to use it's namespace/name in an messaging
+	// see if we have quota for this namespace/project
+
+	// process quota
+	quotaList := quotav1.ClusterResourceQuotaList{}
+	//TODO controller runtime seemed unable to deal with openshift API and its attempt at mapping to CRDs; so for now
+	// we are using a non caching client; as such, we may not be able to afford to this list on every pass through this
+	// reconciler, but we need to get a sense for how often these quotas get defined in appstudio, when they are defined, etc.
+	qerr := r.nonCachingClient.List(ctx, &quotaList)
+	if qerr != nil {
+		return ctrl.Result{}, qerr
+	}
+	for _, quota := range quotaList.Items {
+		// find applicable quota for this namespace
+		ns, ok := quota.Spec.Selector.AnnotationSelector["openshift.io/requester"]
+
+		if !ok {
+			continue
+		}
+		if ns != request.Namespace {
+			continue
+		}
+		hardPodCount := 0
+		if quota.Spec.Quota.Hard.Pods() != nil {
+			hardPodCount = int(quota.Spec.Quota.Hard.Pods().Value())
+		}
+
+		if hardPodCount <= 0 {
+			var quant resource.Quantity
+			quant, ok = quota.Spec.Quota.Hard[corev1.ResourceName("count/pods")]
+			if ok {
+				hardPodCount = int(quant.Value())
+			}
+		}
+		if hardPodCount <= 0 {
+			continue
+		}
+
+		// see how close we are to quota
+		twList := v1alpha1.TektonWrapperList{}
+		if err := r.client.List(ctx, &twList); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		totalCount := len(twList.Items)
+		activeCount := 0
+		throttledCount := 0
+		unprocessedCount := 0
+		doneCount := 0
+		for _, t := range twList.Items {
```

Comment:
Don't we need to check the namespace here?

Reply:
ah shoot - another blind spot since I'm used to vanilla k8s clients (where the namespace filter is baked in) vs. this controller-runtime wrappering

I need to pass in a 

```
opts := client.ListOptions{
		Namespace:     tw.Namespace,
}
```

to the `List(...)` call.  Will update in next push.